nativelib/src/main/resources/gc/immix/datastructures/Bytemap.h

```diff
@@ -0,0 +1,114 @@
+    bm_marked = 0x4,
+} Flag;
+
+void Bytemap_Init(Bytemap *bytemap, word_t *firstAddress, size_t size);
+
+static inline size_t Bytemap_index(Bytemap *bytemap, word_t *address) {
+    size_t index =
+        (address - bytemap->firstAddress) / ALLOCATION_ALIGNMENT_WORDS;
+    assert(address >= bytemap->firstAddress);
+    assert(index < bytemap->size);
+    assert(((word_t)address & ALLOCATION_ALIGNMENT_INVERSE_MASK) ==
+           (word_t)address);
+    return index;
+}
+
+static inline ubyte_t *Bytemap_Cursor(Bytemap *bytemap, word_t *address) {
+    size_t index =
+        (address - bytemap->firstAddress) / ALLOCATION_ALIGNMENT_WORDS;
+    assert(address >= bytemap->firstAddress);
+    assert(index < bytemap->size);
+    assert(((word_t)address & ALLOCATION_ALIGNMENT_INVERSE_MASK) ==
+           (word_t)address);
+    return &bytemap->data[index];
+}
+
+static inline bool Bytemap_IsFree(Bytemap *bytemap, word_t *address) {
+    return bytemap->data[Bytemap_index(bytemap, address)] == bm_free;
+}
+
+static inline bool Bytemap_IsPlaceholder(Bytemap *bytemap, word_t *address) {
+    return bytemap->data[Bytemap_index(bytemap, address)] == bm_placeholder;
+}
+
+static inline bool Bytemap_IsAllocated(Bytemap *bytemap, word_t *address) {
+    return bytemap->data[Bytemap_index(bytemap, address)] == bm_allocated;
+}
+
+static inline bool Bytemap_IsMarked(Bytemap *bytemap, word_t *address) {
+    return bytemap->data[Bytemap_index(bytemap, address)] == bm_marked;
+}
+
+static inline void Bytemap_SetFree(Bytemap *bytemap, word_t *address) {
+    bytemap->data[Bytemap_index(bytemap, address)] = bm_free;
+}
+
+static inline void Bytemap_SetPlaceholder(Bytemap *bytemap, word_t *address) {
+    bytemap->data[Bytemap_index(bytemap, address)] = bm_placeholder;
+}
+
+static inline void Bytemap_SetAllocated(Bytemap *bytemap, word_t *address) {
+    bytemap->data[Bytemap_index(bytemap, address)] = bm_allocated;
+}
+
+static inline void Bytemap_SetMarked(Bytemap *bytemap, word_t *address) {
+    bytemap->data[Bytemap_index(bytemap, address)] = bm_marked;
+}
+
+static inline ubyte_t *Bytemap_NextLine(ubyte_t *cursor) {
+    return cursor + WORDS_IN_LINE / ALLOCATION_ALIGNMENT_WORDS;
+}
+
+static inline void Bytemap_ClearLineAt(ubyte_t *cursor) {
+    memset(cursor, 0, WORDS_IN_LINE / ALLOCATION_ALIGNMENT_WORDS);
+}
+
+static inline void Bytemap_ClearBlock(Bytemap *bytemap, word_t *start) {
+    memset(Bytemap_Cursor(bytemap, start), 0,
+           WORDS_IN_BLOCK / ALLOCATION_ALIGNMENT_WORDS);
+}
+
+/*
+    implements this, just with hardcoded constants:
+
+    size_t startIndex = Bytemap_index(bytemap, start);
+    size_t endIndex = startIndex + WORDS_IN_LINE / ALLOCATION_ALIGNMENT_WORDS;
+    ubyte_t *data = bytemap->data;
+    for (size_t i = startIndex; i < endIndex; i++) {
+        if (data[i] == bm_marked) {
+            data[i] = bm_allocated;
+        } else if (data[i] == bm_allocated) {
+            data[i] = bm_free;
+        }
+    }
+*/
```

Comment:
This comment needs to go inside the method, and use `//` instead of `/* ... */`.